### PR TITLE
Get clientId and issuer separately in the GET application response

### PR DESCRIPTION
**Pull request by ravinduramesh**
### Proposed changes in this pull request
Retrieve clientId and SAML issuer to the frontend to show them on the application listing page. And unit test implementations for GET applications by clientId filter.

### Related Issues
Fixes [#11459](https://github.com/wso2-enterprise/asgardeo-product/issues/11459), [#13933](https://github.com/wso2/product-is/issues/13933)

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 1]
```diff
@@ -350,27 +341,78 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
...
+        inputSP.setApplicationName(APPLICATION_NAME_1);
+        addApplicationConfigurations(inputSP);
 
         // Adding application.
-        applicationManagementService.createApplication(inputSP1, SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
-        applicationManagementService.createApplication(inputSP2, SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
+        String resourceId = applicationManagementService.createApplication(inputSP,
+                SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
+
+        // Retrieving application by ResourceId
```
Comment (kayathiri4) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:370]:
```suggestion
        // Retrieving application by ResourceId.
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 2]
```diff
@@ -350,27 +341,78 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
     }
 
     @Test
-    public void testGetApplicationBasicInfoFilterOffsetLimit() throws IdentityApplicationManagementException {
+    public void testGetApplicationBasicInfoNameFilterOffsetLimit() throws IdentityApplicationManagementException {
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:344]: ByNameFilter...

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 3]
```diff
@@ -350,27 +341,78 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
...
+        // Deleting all added applications.
+        applicationManagementService.deleteApplications(SUPER_TENANT_ID);
+    }
 
-        ServiceProvider inputSP2 = new ServiceProvider();
-        inputSP2.setApplicationName(APPLICATION_NAME_2);
-        addApplicationConfigurations(inputSP2);
+    private String createApplicationWithInboundConfig() throws IdentityApplicationManagementException {
+
+        ServiceProvider inputSP = new ServiceProvider();
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:362]: name the variable as serviceProvider

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 4]
```diff
@@ -350,27 +341,78 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
...
+
+        // Retrieving application by ResourceId.
+        ServiceProvider expectedSP = applicationManagementService.getApplicationByResourceId(resourceId,
+                SUPER_TENANT_DOMAIN_NAME);
+        String addedInboundKey = expectedSP.getInboundAuthenticationConfig().
+                getInboundAuthenticationRequestConfigs()[0].getInboundAuthKey();
+        return addedInboundKey;
+    }
+
+    public void testGetApplicationBasicInfoClientIdFilterOffsetLimit() throws IdentityApplicationManagementException {
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:378]: ByClientIdFilter

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 5]
```diff
@@ -93,8 +93,12 @@ public class ApplicationManagementServiceImplTest extends PowerMockTestCase {
     private static final String SAMPLE_TENANT_DOMAIN = "tenant domain";
     private static final String APPLICATION_NAME_1 = "Test application1";
     private static final String APPLICATION_NAME_2 = "Test application2";
-    private static final String APPLICATION_NAME_1_Filter = "name ew application1";
-    private static final String APPLICATION_NAME_2_Filter = "name co 2";
+    private static final String APPLICATION_NAME_1_FILTER = "name ew application1";
+    private static final String APPLICATION_NAME_2_FILTER = "name co 2";
+    private static final String APPLICATION_CLIENT_ID_FILTER = "clientId co %s";
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:98]: Duplicate same filters for SAML inbound auth type also

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 6]
```diff
@@ -350,27 +341,78 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
...
+
+    public void testGetApplicationBasicInfoClientIdFilterOffsetLimit() throws IdentityApplicationManagementException {
+
+        String addedInboundKey = createApplicationWithInboundConfig();
 
         ApplicationBasicInfo[] applicationBasicInfo1 = applicationManagementService.getApplicationBasicInfo
-                (SUPER_TENANT_DOMAIN_NAME, USERNAME_1, APPLICATION_NAME_1_Filter, 0, 10);
+                (SUPER_TENANT_DOMAIN_NAME, USERNAME_1, String.format(APPLICATION_CLIENT_ID_FILTER, addedInboundKey),
+                        0, 10);
         Assert.assertEquals(applicationBasicInfo1[0].getApplicationName(), APPLICATION_NAME_1);
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:385]: Use data provider to add random applications

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 7]
```diff
@@ -350,27 +341,78 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
...
+
+        String addedInboundKey = createApplicationWithInboundConfig();
+
+        ApplicationBasicInfo[] applicationBasicInfo1 = applicationManagementService.getApplicationBasicInfo
+                (SUPER_TENANT_DOMAIN_NAME, USERNAME_1, String.format(APPLICATION_NAME_OR_CLIENT_ID_FILTER,
+                        addedInboundKey), 0, 10);
+        Assert.assertEquals(applicationBasicInfo1[0].getApplicationName(), APPLICATION_NAME_1);
+
+        // Deleting all added applications.
+        applicationManagementService.deleteApplications(SUPER_TENANT_ID);
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:403]: creating and deleting SP in every function is unnecessary.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 8]
```diff
@@ -388,25 +430,53 @@ public void testGetCountOfAllApplications() throws IdentityApplicationManagement
...
 
-        // Adding application.
-        applicationManagementService.createApplication(inputSP1, SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
-        applicationManagementService.createApplication(inputSP2, SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
+
+        // Deleting all added applications.
+        applicationManagementService.deleteApplications(SUPER_TENANT_ID);
+    }
+
+    public void testGetCountOfApplicationsClientIdFilter() throws IdentityApplicationManagementException {
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:447]: Can combine following 3 test function into 1

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 9]
```diff
@@ -93,8 +93,12 @@ public class ApplicationManagementServiceImplTest extends PowerMockTestCase {
     private static final String SAMPLE_TENANT_DOMAIN = "tenant domain";
     private static final String APPLICATION_NAME_1 = "Test application1";
     private static final String APPLICATION_NAME_2 = "Test application2";
-    private static final String APPLICATION_NAME_1_Filter = "name ew application1";
-    private static final String APPLICATION_NAME_2_Filter = "name co 2";
+    private static final String APPLICATION_NAME_1_FILTER = "name ew application1";
+    private static final String APPLICATION_NAME_2_FILTER = "name co 2";
```
Comment (ravinduramesh) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:97]: check whether we can remove these with data provider or change var name to application_name_filter_2

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 10]
```diff
@@ -5200,10 +5198,20 @@ private ApplicationBasicInfo buildApplicationBasicInfoWithInboundConfig(ResultSe
         basicInfo.setApplicationId(appNameResultSet.getInt(ApplicationTableColumns.ID));
         basicInfo.setApplicationName(appNameResultSet.getString(ApplicationTableColumns.APP_NAME));
         basicInfo.setDescription(appNameResultSet.getString(ApplicationTableColumns.DESCRIPTION));
+
```
Comment (kayathiri4) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:5201]: Do we need this additional line?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 11]
```diff
@@ -5200,10 +5198,20 @@ private ApplicationBasicInfo buildApplicationBasicInfoWithInboundConfig(ResultSe
...
         basicInfo.setApplicationResourceId(appNameResultSet.getString(ApplicationTableColumns.UUID));
         basicInfo.setImageUrl(appNameResultSet.getString(ApplicationTableColumns.IMAGE_URL));
         basicInfo.setAccessUrl(appNameResultSet.getString(ApplicationTableColumns.ACCESS_URL));
-        basicInfo.setInboundKey(appNameResultSet.getString(ApplicationInboundTableColumns.INBOUND_AUTH_KEY));
+
+        String inboundAuthKey = appNameResultSet.getString(ApplicationInboundTableColumns.INBOUND_AUTH_KEY);
+        String inboundAuthType = appNameResultSet.getString(ApplicationInboundTableColumns.INBOUND_AUTH_TYPE);
+        if (StringUtils.isNotBlank(inboundAuthKey)) {
+            if (inboundAuthType.equals("oauth2")) {
```
Comment (kayathiri4) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:5209]: Can we use the constants?

**Review by kayathiri4 — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java [hunk 12]
```diff
@@ -179,23 +180,43 @@ public void setAppOwner(User appOwner) {
     }
 
     /**
-     * Get inbound auth key.
+     * Get oAuth2 client ID.
```
Comment (inthirakumaaran) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java:183]: It is `OAuth2` not `oAtuh2`, please check the other places as well

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 13]
```diff
@@ -2034,9 +2043,18 @@ public ServiceProvider getApplicationWithRequiredAttributes(int applicationId, L
         try {
             // Load basic application data
             ServiceProvider serviceProvider = getBasicApplicationData(applicationId, connection);
-            if (CollectionUtils.isNotEmpty(requiredAttributes)) {
+            int tenantID = IdentityTenantUtil.getTenantId(serviceProvider.getTenantDomain());
+
+            // Avoid iterating over similar items in requiredAttributes list.
+            Set<String> requiredAttributesSet = new HashSet<>(requiredAttributes);
+            if (requiredAttributesSet.contains(CLIENT_ID_SP_PROPERTY_NAME) &&
```
Comment (inthirakumaaran) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2050]: why do we need to check for `CLIENT_ID_SP_PROPERTY_NAME` here?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 14]
```diff
@@ -5196,14 +5227,28 @@ private ApplicationBasicInfo buildApplicationBasicInfo(ResultSet appNameResultSe
...
+
         basicInfo.setApplicationResourceId(appNameResultSet.getString(ApplicationTableColumns.UUID));
         basicInfo.setImageUrl(appNameResultSet.getString(ApplicationTableColumns.IMAGE_URL));
         basicInfo.setAccessUrl(appNameResultSet.getString(ApplicationTableColumns.ACCESS_URL));
-        basicInfo.setInboundKey(appNameResultSet.getString(ApplicationInboundTableColumns.INBOUND_AUTH_KEY));
+
+        String inboundAuthKey = appNameResultSet.getString(ApplicationInboundTableColumns.INBOUND_AUTH_KEY);
+        String inboundAuthType = appNameResultSet.getString(ApplicationInboundTableColumns.INBOUND_AUTH_TYPE);
+        if (StringUtils.isNotBlank(inboundAuthKey)) {
+            if (inboundAuthType.equals("oauth2")) {
```
Comment (inthirakumaaran) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:5246]: Don't we already have constants for "oauth2" or "samlsso" ?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 15]
```diff
@@ -350,27 +343,59 @@ public void testGetApplicationBasicInfoOffsetLimit() throws IdentityApplicationM
...
 
         // Adding application.
         applicationManagementService.createApplication(inputSP1, SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
         applicationManagementService.createApplication(inputSP2, SUPER_TENANT_DOMAIN_NAME, USERNAME_1);
 
-        ApplicationBasicInfo[] applicationBasicInfo1 = applicationManagementService.getApplicationBasicInfo
-                (SUPER_TENANT_DOMAIN_NAME, USERNAME_1, APPLICATION_NAME_1_Filter, 0, 10);
-        Assert.assertEquals(applicationBasicInfo1[0].getApplicationName(), APPLICATION_NAME_1);
+        // Test get applications with name filter.
+        ApplicationBasicInfo[] applicationBasicInfo = applicationManagementService.getApplicationBasicInfo
```
Comment (inthirakumaaran) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:363]: Cannot we abstract this logic? IMO we can use dataproviders here

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java (hunk 12 above)
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java:183]: Fixed in https://github.com/wso2/carbon-identity-framework/pull/4432

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 16]
```diff
@@ -1923,9 +1925,16 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
 
             appNameResultSet = getAppNamesStmt.executeQuery();
 
-            while (appNameResultSet.next()) {
-                appInfo.add(buildApplicationBasicInfoWithInboundConfig(appNameResultSet));
+            if (filterString.contains("SP_INBOUND_AUTH")) {
```
[the remark this answers is not in the document]
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:1928]: As it's used only in a single place, do we need to create a public constant for this?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 13 above)
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2050]: The logic here is to remove one if both are there in the requiredAttributesSet variable.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java (hunk 15 above)
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:363]: Resolved with https://github.com/wso2/carbon-identity-framework/pull/4452

**Pull request merged**